code/framework/object/src/main/java/io/cattle/platform/object/meta/impl/DefaultObjectMetaDataManager.java

```diff
@@ -683,6 +683,25 @@ public Relationship getRelationship(String type, String linkName) {
     }
 
     @Override
+    public Relationship getRelationship(String type, String linkName, String fieldName) {
+        Class<?> clz = schemaFactory.getSchemaClass(type, true);
+        Map<String, List<Relationship>> relationship = relationshipsBothCase.get(clz);
+        List<Relationship> relationshipList = relationship.get(linkName);
+        if (relationshipList == null || relationshipList.size() == 0) {
+            return null;
+        }
+        if (relationshipList.size() > 1) {
+            for (Relationship rel : relationshipList) {
+                if (rel instanceof ForeignKeyRelationship && rel.getName() != null && rel.getName().equalsIgnoreCase(linkName) && rel.getPropertyName().equalsIgnoreCase(fieldName)) {
+                    return rel;
+                }
+            }
+            return null;
+        }
```

Comment:
I'm good going either way.  Prior to my changes, cattle didn't check since there was just 1 Relationship rather than a List<Relationship>.  With one, there's no ambiguity.  When I changed it to a List, I tried to keep as much of the original behavior as possible since I was new to the codebase (and it'd instill more confidence in the code reviewer).